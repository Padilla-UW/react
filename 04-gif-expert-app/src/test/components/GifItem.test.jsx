import { render, screen } from "@testing-library/react"
import { GifItem } from "../../components"

describe('Pruebas en GifItem',()=>{
    const title = 'one punch';
    const url = 'http://one-punch.com/img.jpg';

    test('Debe hacer match con el snapshot', () => { 
        const {container} = render(<GifItem 
            title={title}
            url={url}  
          />)
          expect(container).toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el url y alt asignados', () => {
        render(<GifItem 
            title={title}
            url={url}  
          />);

        const { src, alt } = screen.getByRole('img');
        //expect(screen.getByRole('img').src).toBe(url);
        expect(src).toBe(url);
        expect(alt).toBe(title);    

     });

     test('Debe de agregar el titulo de la imagen',()=>{
        render(<GifItem 
            title={title}
            url={url}  
          />);
        expect(screen.getByText(title)).toBeTruthy();
     })
})