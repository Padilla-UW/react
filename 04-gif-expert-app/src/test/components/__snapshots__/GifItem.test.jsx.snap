// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="one punch"
      src="http://one-punch.com/img.jpg"
    />
    <p>
      one punch
    </p>
  </div>
</div>
`;
